packages/web/stores/imperator-queries/pools.ts

```diff
@@ -0,0 +1,104 @@
+import { KVStore } from "@keplr-wallet/common";
+import {
+  ObservableQuery,
+  QueryOptions,
+  QueryResponse,
+} from "@keplr-wallet/stores";
+import Axios, { AxiosInstance, CancelToken } from "axios";
+import { makeObservable, observable, action, computed } from "mobx";
+import { computedFn } from "mobx-utils";
+import { PricePretty, CoinPretty, Dec, IntPretty } from "@keplr-wallet/unit";
+
+import { ObservablePool } from "@osmosis-labs/stores";
+
+import { IMPERATOR_API_DOMAIN } from "../../constants";
+import { FiatCurrency } from "@keplr-wallet/types";
+
+interface PoolMetric {
+  pool_id: string;
+  volume_24h: number;
+  volume_7d: number;
+  fees_spent_24h: number;
+  fees_spent_7d: number;
+  fees_percentage: number;
+}
+
+export interface ObservablePoolWithMetric {
+  pool: ObservablePool;
+  liquidity: PricePretty;
+  volume24h: PricePretty;
+  fees7d: PricePretty;
+  myLiquidity?: PricePretty;
+  epochsRemaining?: number;
+  apr?: string;
+}
+
+export class ObservableImperatorQuery<
+  T = unknown,
+  E = unknown
+> extends ObservableQuery<T, E> {
+  constructor(kvStore: KVStore, urlPath: string) {
+    const instance = Axios.create({ baseURL: IMPERATOR_API_DOMAIN });
+
+    super(kvStore, instance, urlPath);
+  }
+}
+
+export class ObservableImperatorQueryPoolMetrics extends ObservableImperatorQuery<{
+  last_update_at: number;
+  data: PoolMetric[];
+}> {
+  constructor(kvStore: KVStore) {
+    super(kvStore, "/fees/v1/pools");
+
+    makeObservable(this);
+  }
+
+  readonly makePoolsWithMetric = computedFn(
```

Comment:
Why does this accept a list of pools? Given that it is memoized I think it would be more general to calculate for one pool at a time. Call it `makePoolFeeMetrics` then it returns `ObservablePoolWithFeeMetrics` object.